friendshipper/src-tauri/src/repo/operations/status.rs

```diff
@@ -405,14 +398,22 @@ where
             .await
             .is_ok();
 
-        if !upstream_exists {
-            return Ok(vec![]);
-        }
-
         let commit_range = format!("HEAD...origin/{}", branch);
 
-        // check for files modified on the upstream trunk branch
-        let modified_upstream: Vec<String> = self.git_client.diff_filenames(&commit_range).await?;
+        // check for files modified on the upstream branch
+        let mut modified_upstream: Vec<String> = match upstream_exists {
+            true => self.git_client.diff_filenames(&commit_range).await?,
+            false => vec![],
+        };
+
+        // check for files modified on the target upstream branches
+        let target_branches = self.repo_config.read().target_branches.clone();
+        for target_branch in target_branches {
+            let commit_range = format!("HEAD...origin/{}", target_branch.name);
+            let modified_in_target_branch: Vec<String> =
+                self.git_client.diff_filenames(&commit_range).await?;
+            modified_upstream.extend(modified_in_target_branch);
```

Comment:
the `user_modified_upstream` vec below would miss out on files that have already been merged between different target branches, so we do that filtering after aggregating all the target branch changes.